go/inst/instance.go

```diff
@@ -293,9 +327,27 @@ func (this *Instance) IsReplica() bool {
 	return this.MasterKey.Hostname != "" && this.MasterKey.Hostname != "_" && this.MasterKey.Port != 0 && (this.ReadBinlogCoordinates.LogFile != "" || this.UsingGTID())
 }
 
-// IsMaster makes simple heuristics to decide whether this instance is a master (not replicating from any other server)
+// IsMaster makes simple heuristics to decide whether this instance is a master (not replicating from any other server),
+// either via traditional async/semisync replication or group replication
 func (this *Instance) IsMaster() bool {
-	return !this.IsReplica()
+	// If traditional replication is configured, it is for sure not a master
+	if this.IsReplica() {
+		return false
+	} else {
+		// If traditional replication is not configured, and it is also not part of a replication group, this host is
+		// a master
+		if !this.IsReplicationGroupMember() {
+			return true
+		} else {
+			// If traditional replication is not configured, and this host is part of a group, it is only considered a
+			// master if it has the role of group Primary. Otherwise it is not a master.
+			if this.ReplicationGroupMemberRole == GroupReplicationMemberRolePrimary {
+				return true
+			} else {
+				return false
+			}
+		}
+	}
```

Comment:
I'm replacing this code:
```golang
func (this *Instance) IsMaster() bool {
	// If traditional replication is configured, it is for sure not a master
	if this.IsReplica() {
		return false
	} else {
		// If traditional replication is not configured, and it is also not part of a replication group, this host is
		// a master
		if !this.IsReplicationGroupMember() {
			return true
		} else {
			// If traditional replication is not configured, and this host is part of a group, it is only considered a
			// master if it has the role of group Primary. Otherwise it is not a master.
			if this.ReplicationGroupMemberRole == GroupReplicationMemberRolePrimary {
				return true
			} else {
				return false
			}
		}
	}
}
```
with simplified nesting:
```golang
func (this *Instance) IsMaster() bool {
	// If traditional replication is configured, it is for sure not a master
	if this.IsReplica() {
		return false
	}
	// If traditional replication is not configured, and it is also not part of a replication group, this host is
	// a master
	if !this.IsReplicationGroupMember() {
		return true
	}
	// If traditional replication is not configured, and this host is part of a group, it is only considered a
	// master if it has the role of group Primary. Otherwise it is not a master.
	if this.ReplicationGroupMemberRole == GroupReplicationMemberRolePrimary {
		return true
	}
	return false
}
```